issues: rework unit test failure template

tbg opened the pull request:
Continuing to make progress towards letting roachtest file issues with a
more appropriate template by introducing template use into
`internal/issues` which made the code vastly more maintainable.

I took the opportunity to streamline the template we use for unit test
failures (used both from the bleeding edge build and nightly stress):

<img width="753" alt="image" src="https://user-images.githubusercontent.com/5076964/70328607-bf4e3e80-1839-11ea-9036-9229690fbca5.png">

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43021)
<!-- Reviewable:end -->

jordanlewis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> nice

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43021#-:-Lvd3jg8264NHzJMqry6:bb76hxo)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jordanlewis)



<!-- Sent from Reviewable.io -->

tbg committed:
issues: rework unit test failure template

Continuing to make progress towards letting roachtest file issues with a
more appropriate template by introducing template use into
`internal/issues` which made the code vastly more maintainable.

I took the opportunity to streamline the template we use for unit test
failures (used both from the bleeding edge build and nightly stress).

Release note: None

tbg commented:
bors r=jordanlewis
